src/sql/src/func.rs

```diff
@@ -4102,19 +4102,42 @@ pub static MZ_INTERNAL_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(
             ") => Oid, oid::FUNC_DATABASE_OID_OID;
         },
         // There is no regclass equivalent for schemas to look up oids, so we have this helper function instead.
-        // TODO: Support qualified names.
-        // TODO: This should take into account the search path when looking for an object.
         "mz_schema_oid" => Scalar {
             params!(String) => sql_impl_func("
-                CASE
+            CASE
                 WHEN $1 IS NULL THEN NULL
-                ELSE (
-                    mz_unsafe.mz_error_if_null(
-                        (SELECT oid FROM mz_schemas WHERE name = $1),
-                        'schema \"' || $1 || '\" does not exist'
-                    )
+            ELSE
+                mz_unsafe.mz_error_if_null(
+                    (
+                        SELECT
+                            CASE
+                                -- Schema names should only have 2 components, so error if there's
+                                -- three.
+                                WHEN n[1] IS NOT NULL THEN
```

Comment:
This took a while to figure out that it's because mz_normalize_object_name always returns an array with 3 items, the left most is indexed as 1, and it's the database. So weird but it works.

Reply:
I'll try and extract this logic into a `normalize_schema_name` function. That should hopefully be more readable and clear things up.